invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/detail.html

```diff
@@ -60,9 +60,9 @@ <h1>{{ metadata.title }}</h1>
         {%- endblock record_content -%}
         {# files #}
         {%- block record_files -%}
-        {%- if record.files and record|can_list_files %}
-          {%- set files = record.files|make_files_preview_compatible %}
-          {%- set default_preview_file = files|select_preview_file(default_preview=record.files.default_preview) %}
+        {%- if files.enabled and record|can_list_files %}
+          {%- set files = files.entries %}
+          {%- set default_preview_file = files|select_preview_file(default_preview=files.default_preview) %}
           {{ preview_file_box(default_preview_file, pid) }}
```

Comment:
i think this should be made conditional, something along the lines of

`{% if any([file is previewable for file in files]) %}`

... just in a valid fashion, using the `previewable` test and filters - does anybody know how to do this?

Reply:
fixed with a new filter, `has_previewable_files`